src/vs/workbench/contrib/notebook/browser/view/cellPart.ts

```diff
@@ -57,49 +64,149 @@ export abstract class CellPart extends Disposable {
 	updateForExecutionState(element: ICellViewModel, e: ICellExecutionStateChangedEvent): void { }
 }
 
+export abstract class CellOverlayPart extends Disposable {
```

Comment:
Please include some description of when to use this vs CellContentPart. It would not be totally clear if I was implementing a new part